code/controllers/subsystem/gravity.dm

```diff
@@ -0,0 +1,88 @@
+var/datum/subsystem/gravity/SSgravity
+
+/datum/subsystem/gravity
+	name = "Gravity"
+	priority = 75
+	wait = 1
+	can_fire = FALSE
+	flags = SS_KEEP_TIMING | SS_BACKGROUND
+
+	var/list/currentrun = list()
+	var/list/processing = list()
+
+	//Only keeps track of station/Z1.
+	var/gravity_direction = FALSE	//FALSE for normal, and cardinals.
+	var/gravity_strength = 1
+	var/calculation_timer = 0
+	var/calculation_interval = 2400
+	var/throwing = FALSE
+	var/recalculation_tick_cost = 0
+	var/gravstun = TRUE
+	var/gravstun_amount = 10
+
+/datum/subsystem/gravity/New()
+	NEW_SS_GLOBAL(SSgravity)
+
+/datum/subsystem/gravity/Initialize()
+	recalculate_atoms()
+
+/datum/subsystem/gravity/proc/set_gravity_direction(dir)
+	can_fire = FALSE
+	if(!dir)
+		gravity_direction = FALSE
+		processing = list()
+		currentrun = list()
+		return
+	gravity_direction = dir
+	recalculate_atoms()
+	if(gravstun)
+		stun_all_mobs()
+	can_fire = TRUE
+	throw_everything()
+
+/datum/subsystem/gravity/proc/stun_all_mobs()
+	for(var/mob/living/carbon/C in world)
+		if(C.z == 1)
+			C.Weaken(gravstun_amount)
+
+/datum/subsystem/gravity/proc/recalculate_atoms()
+	processing = list()
+	var/before = world.time
+	for(var/atom/movable/A in world)
```

Comment:
it should be world.time - before, otherwise your values would be negative.
also taking no ticks doesn't mean it isn't eating up the entirety of the tick it does run in.